File: Dev/C++/Mascaret/src/VEHA/Behavior/StateMachine/StateMachine.cpp
#include "VEHA/Kernel/InstanceSpecification.h"
#include "VEHA/Behavior/StateMachine/StateMachine.h"
#include "VEHA/Behavior/StateMachine/Region.h"
#include "VEHA/Behavior/StateMachine/StateMachineBehaviorExecution.h"
#include "VEHA/Behavior/Common/BehaviorExecution.h"

namespace VEHA
{


StateMachine::StateMachine(const string& name)
:Behavior(name)
{
}

StateMachine::~StateMachine()
{
}

shared_ptr<BehaviorExecution> StateMachine::createBehaviorExecution(shared_ptr<InstanceSpecification> host,const Parameters& p, bool sync)
{
	shared_ptr<StateMachineBehaviorExecution> behavior(new StateMachineBehaviorExecution(shared_dynamic_cast<StateMachine>(shared_from_this()),host,p));
	host->addStateMachineExecution(behavior);
	return behavior;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/StateMachine/StateMachine.cs

using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class StateMachine : Behavior
    {


        private List<Region> region = new List<Region>();
        public List<Region> Region
        {
            get { return region; }
            set { region = value; }
        }


        private List<PseudoState> connectionPoint = new List<PseudoState>();
        public List<PseudoState> ConnectionPoint
        {
            get { return connectionPoint; }
            set { connectionPoint = value; }
        }


        private List<State> submachinestates = new List<State>();
        public List<State> Submachinestates
        {
            get { return submachinestates; }
            set { submachinestates = value; }
        }

        public StateMachine(string name)
            : base(name)
        {
        }

        public override BehaviorExecution createBehaviorExecution(InstanceSpecification host, Dictionary<string, ValueSpecification> p, bool sync)
        {
            StateMachineBehaviorExecution behavior = new StateMachineBehaviorExecution(this, host, p, sync);
            host.SmBehaviorExecutions.Add(behavior);
            return behavior;
        }

    }
}